libs/community/langchain_community/chains/pebblo_retrieval/utilities.py

```diff
@@ -64,3 +85,313 @@ def get_ip() -> str:
+        return headers
+
+    @staticmethod
+    def make_request(
+        method: str,
+        url: str,
+        headers: dict,
+        payload: Optional[dict] = None,
+        timeout: int = 20,
+    ) -> Optional[Response]:
+        """
+        Make a request to the Pebblo server/cloud API.
+
+        Args:
+            method (str): HTTP method (GET, POST, PUT, DELETE, etc.).
+            url (str): URL for the request.
+            headers (dict): Headers for the request.
+            payload (Optional[dict]): Payload for the request (for POST, PUT, etc.).
+            timeout (int): Timeout for the request in seconds.
+
+        Returns:
+            Optional[Response]: Response object if the request is successful.
+        """
+        try:
+            response = request(
+                method=method, url=url, headers=headers, json=payload, timeout=timeout
+            )
+            logger.debug(
+                "Request: method %s, url %s, body %s len %s response status %s body %s",
+                method,
+                response.request.url,
+                str(response.request.body),
+                str(len(response.request.body if response.request.body else [])),
+                str(response.status_code),
+                response.json(),
+            )
+
+            if response.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
+                logger.warning(f"Pebblo Server: Error {response.status_code}")
+            elif response.status_code >= HTTPStatus.BAD_REQUEST:
+                logger.warning(f"Pebblo received an invalid payload: {response.text}")
+            elif response.status_code != HTTPStatus.OK:
+                logger.warning(
+                    f"Pebblo returned an unexpected response code: "
+                    f"{response.status_code}"
+                )
+
+            return response
+        except RequestException:
+            logger.warning("Unable to reach server %s", url)
+        except Exception as e:
+            logger.warning("An Exception caught in make_request: %s", e)
+        return None
+
+    @staticmethod
+    def update_cloud_payload(payload: dict, pebblo_resp: Optional[Response]) -> None:
+        """
+        Update the payload with response, prompt and context from Pebblo response.
+
+        Args:
+            payload (dict): Payload to be updated.
+            pebblo_resp (Optional[Response]): Response from Pebblo server.
+        """
+        if pebblo_resp:
+            resp = json.loads(pebblo_resp.text)
+            if resp:
+                payload["response"].update(
+                    resp.get("retrieval_data", {}).get("response", {})
+                )
+                payload["response"].pop("data")
+                payload["prompt"].update(
+                    resp.get("retrieval_data", {}).get("prompt", {})
+                )
+                payload["prompt"].pop("data")
+                context = payload["context"]
```

Comment:
We should use get here in  payload["response"], payload["prompt"], payload["context"]